src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    // перевірка, чи будь-яка зі сторін <= 0
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Sides of triangle should be greater than zero.');
+    }
+
+    // перевірка, чи найдовша сторона >= двом іншим
+    if (
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.a + this.b
+    ) {
+      throw new Error(
+        `sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
+      );
+    }
+    this.shape = 'triangle';
```

Comment:
The 'shape' property is being manually set to 'triangle' which is redundant because its type is already declared as a literal type 'triangle' on line 11. You can remove this assignment. (checklist item 4)